### DISCUSSION TITLE
"frankenphp php-server" vs "frankenphp run" when to use what?

### DISCUSSION THREAD

**Discussion by ochorocho** [php/frankenphp#469] [Q&A] [answered]
Hey, 

just had a little play with FrankenPHP locally (self compiled).

So my App is working using the following command and caddyfile:

```
./frankenphp-mac-arm64 run -c franken-caddy/Caddyfile
```
[Caddyfile example](https://github.com/dunglas/frankenphp/files/13894069/Caddyfile.txt)

Then i discovered the `php-server` subcommand, where my app is not working because it needs a custom Caddy file:
```
./frankenphp-mac-arm64 php-server -r ./ -l 127.0.0.1:8788 --debug
```

**What is the difference between using `run` and `php-server`, and what is the recommended way of running FrankenPHP?**

**Comment by dunglas** [comment, marked as answer]
`php-server` is a helper command that starts FrankenPHP without requiring to write a Caddyfile. It is ok for very standard PHP apps. It doesn't read the `Caddyfile` at all.

If you need (or want) a custom Caddyfile, you must use `run` instead.
